pulsar-broker/src/test/java/org/apache/pulsar/broker/LedgerLostTest.java

```diff
@@ -0,0 +1,429 @@
+        pulsarConfig.setLazyCursorRecovery(false);
+    }
+
+    protected void startPulsar() throws Exception {
+        log.info("===> Start pulsar ");
+        pulsarService = new PulsarService(pulsarConfig);
+        pulsarService.start();
+        brokerWebServicePort = pulsarService.getListenPortHTTP().get();
+        brokerServicePort = pulsarService.getBrokerListenPort().get();
+        brokerUrl = String.format("http://%s:%s", LOCALHOST, brokerWebServicePort);
+        brokerServiceUrl = String.format("pulsar://%s:%s", LOCALHOST, brokerServicePort);
+        initPulsarAdmin();
+        initPulsarClient();
+        initDefaultNamespace();
+    }
+
+    protected void silentStopPulsar() throws Exception {
+        log.info("===> Close pulsar client ");
+        try {
+            pulsarClient.close();
+        }catch (Exception e){
+            log.error("===> Close pulsar client fail", e);
+        }
+        log.info("===> Close pulsar admin ");
+        try {
+            pulsarAdmin.close();
+        }catch (Exception e){
+            log.error("===> Close pulsar admin fail", e);
+        }
+        log.info("===> Stop pulsar service ");
+        try {
+            pulsarService.close();
+        }catch (Exception e){
+            log.error("===> Stop pulsar service fail", e);
+        }
+    }
+
+    protected void stopPulsar() throws Exception {
+        log.info("===> Close pulsar client ");
+        pulsarClient.close();
+        log.info("===> Close pulsar admin ");
+        pulsarAdmin.close();
+        log.info("===> Stop pulsar service ");
+        pulsarService.close();
+    }
+
+    protected void initPulsarAdmin() throws Exception {
+        pulsarAdmin = PulsarAdmin.builder().serviceHttpUrl(brokerUrl).build();
+    }
+
+    protected void initPulsarClient() throws Exception {
+        pulsarClient = PulsarClient.builder().serviceUrl(brokerServiceUrl).build();
+    }
+
+    protected void initDefaultNamespace() throws Exception {
+        if (!pulsarAdmin.clusters().getClusters().contains(CLUSTER)) {
+            pulsarAdmin.clusters().createCluster(CLUSTER, ClusterData.builder().serviceUrl(brokerServiceUrl).build());
+        }
+        if (!pulsarAdmin.tenants().getTenants().contains(DEFAULT_TENANT)){
+            pulsarAdmin.tenants().createTenant(DEFAULT_TENANT,
+                    TenantInfo.builder().allowedClusters(Collections.singleton(CLUSTER)).build());
+        }
+        if (!pulsarAdmin.namespaces().getNamespaces(DEFAULT_TENANT).contains(DEFAULT_NAMESPACE)) {
+            pulsarAdmin.namespaces().createNamespace(DEFAULT_NAMESPACE, Collections.singleton(CLUSTER));
+        }
+    }
+
+    @DataProvider(name = "batchEnabled")
+    public Object[][] batchEnabled(){
+        return new Object[][]{
+                {true},
+                {false}
+        };
+    }
+
+    @Test(timeOut = 30000, dataProvider = "batchEnabled")
+    public void testCompactionLedgerLost(boolean enabledBatch) throws Exception {
```

Comment:
Could you explain why `compaction ledger`?

Reply:
It's a misnomer, I copied it from somewhere and forgot to change it. already fixed.